webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/BinaryDataController.scala

```diff
@@ -443,6 +455,74 @@ class BinaryDataController @Inject()(
       image
     }
 
+  private def findPositionWithData(organizationName: String, dataSetName: String, dataLayerName: String)(
+      implicit m: MessagesProvider) =
+    for {
+      (dataSource, dataLayer) <- getDataSourceAndDataLayer(organizationName, dataSetName, dataLayerName)
+      position <- checkAllPositionsForData(dataSource, dataLayer)
+    } yield position
+
+  private def checkAllPositionsForData(dataSource: DataSource, dataLayer: DataLayer) = {
+    def positionIter(positions: List[Point3D]): Fox[Point3D] =
+      positions match {
+        case List() => Fox.empty
+        case head :: tail =>
+          checkIfPositionHasData(head).futureBox.flatMap {
+            case Full(pos) => Fox.successful(pos)
+            case _         => positionIter(tail)
+          }
+      }
+
+    def checkIfPositionHasData(position: Point3D) = {
+      val request = DataRequest(
+        new VoxelPosition(position.x, position.y, position.z, dataLayer.lookUpResolution(0)),
+        DataLayer.bucketLength,
+        DataLayer.bucketLength,
+        DataLayer.bucketLength
+      )
+      for {
+        (data, _) <- requestData(dataSource, dataLayer, request)
+        if data.nonEmpty
```

Comment:
This syntax is new to me, can we get rid of bool2Fox this way?
However, nonEmpty does not suffice, since the array is still nonEmpty if it contains 32768 zeroes. And indeed, the route always returns the center of the bounding box for me